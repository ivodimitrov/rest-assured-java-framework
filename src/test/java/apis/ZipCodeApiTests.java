package apis;

import org.approvaltests.Approvals;
import org.testng.annotations.Test;

public class ZipCodeApiTests {

    @Test
    public void testWhiteHouseZipBody() {
        ZipCodeApi api = new ZipCodeApi("20500");
        api.get().then().statusCode(200);
        Approvals.verify(api.getBody());
    }

    @Test
    public void testWhiteHouseZipAll() {
        ZipCodeApi api = new ZipCodeApi("20500");
        api.get();
        Approvals.verify(api.getFullResponse(true));
    }
}