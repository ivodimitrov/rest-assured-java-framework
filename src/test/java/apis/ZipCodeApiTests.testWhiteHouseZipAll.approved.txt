#####DATE
Content-Type=application/json
Transfer-Encoding=chunked
Connection=keep-alive
x-cache=hit
charset=UTF-8
vary=Accept-Encoding
CF-Cache-Status=DYNAMIC
#####Report-To
NEL={"success_fraction":0,"report_to":"cf-nel","max_age":604800}
access-control-allow-origin=*
Server=cloudflare
#####CF-RAY
Content-Encoding=gzip
alt-svc=h3=":443"; ma=86400, h3-29=":443"; ma=86400

{
    "post code": "20500",
    "country": "United States",
    "country abbreviation": "US",
    "places": [
        {
            "place name": "Washington",
            "longitude": "-77.0355",
            "state": "District of Columbia",
            "state abbreviation": "DC",
            "latitude": "38.8946"
        }
    ]
}